kratos/geometries/geometry.h

```diff
@@ -832,12 +832,72 @@ class Geometry : public PointerVector<TPointType>
         return normal;
     }
 
+    /**
+     * @brief It returns the vector, which is normal to its corresponding
+     *        geometry in the given integration point for the default
+     *        integration method.
+     * @param IntegrationPointIndex index in internal integration point list
+     * @return The normal in the given integration point
+     */
+    virtual array_1d<double, 3> Normal(
+        IndexType IntegrationPointIndex) const
+    {
+        return Normal(IntegrationPointIndex, mpGeometryData->DefaultIntegrationMethod());
+    }
+
+    /**
+     * @brief It returns the vector, which is normal to its corresponding
+     *        geometry in the given integration point.
+     * @param IntegrationPointIndex index in internal integration point list
+     * @param ThisMethod the integration point is dependent on the used
+     *        integration method
+     * @return The normal in the given integration point
+     */
+    virtual array_1d<double, 3> Normal(
+        IndexType IntegrationPointIndex,
+        IntegrationMethod ThisMethod) const
+    {
+        const SizeType local_space_dimension = this->LocalSpaceDimension();
+        const SizeType dimension = this->WorkingSpaceDimension();
+
+        KRATOS_DEBUG_ERROR_IF(dimension == local_space_dimension)
+            << "Remember the normal can be computed just in geometries with a local dimension: "
+            << this->LocalSpaceDimension() << "smaller than the spatial dimension: "
+            << this->WorkingSpaceDimension() << std::endl;
+
+        // We define the normal and tangents
+        array_1d<double, 3> tangent_xi(3, 0.0);
```

Comment:
```suggestion
        array_1d<double, 3> tangent_xi = ZeroVector(3);
```
This initialization will stop working with AMatrix